### ISSUE TITLE
Rubber Duck 2.2.0 failure to install via local admin account?

### ISSUE THREAD

**Issue by yosso** [rubberduck-vba/Rubberduck#3922]
Our IT department setup a local ADMIN account on my box due to my need to "experiment".

I right-click on the installer and select install as admin and use my local admin login.

Nothing is showing up in the VBA add-in dialog in Excel 2018 (32 bit).  I did uninstall the older versions before trying the install.

The install log 

```
2018-04-11 08:00:31.298   Log opened. (Time zone: UTC-05:00)
2018-04-11 08:00:31.298   Setup version: Inno Setup version 5.5.9 (u)
2018-04-11 08:00:31.298   Original Setup EXE: C:\Users\msjones\Desktop\TEMP\Rubberduck.Setup.2.2.0.exe
2018-04-11 08:00:31.298   Setup command line: /SL5="$711BC,3520412,121344,C:\Users\msjones\Desktop\TEMP\Rubberduck.Setup.2.2.0.exe" 
2018-04-11 08:00:31.298   Windows version: 10.0.16299  (NT platform: Yes)
2018-04-11 08:00:31.298   64-bit Windows: Yes
2018-04-11 08:00:31.298   Processor architecture: x64
2018-04-11 08:00:31.298   User privileges: None
2018-04-11 08:00:31.302   64-bit install mode: Yes
2018-04-11 08:00:37.635   Created temporary directory: c:\temp\is-D9C0T.tmp
2018-04-11 08:00:37.639   .Net v4.5 Framework was found on the system
2018-04-11 08:00:37.707   HasElevateSwitch: 0
2018-04-11 08:00:37.707   IsElevated: 0
2018-04-11 08:00:37.712   PageSkipped set to true now
2018-04-11 08:00:37.712   PageSkipped set to true now
2018-04-11 08:00:39.757   Message box (Yes/No):
                          Setup is not complete. If you exit now, the program will not be installed.
                          
                          You may run Setup again at another time to complete the installation.
                          
                          Exit Setup?
2018-04-11 08:00:42.322   User chose Yes.
2018-04-11 08:00:42.322   Deinitializing Setup.
2018-04-11 08:00:42.336   Log closed.
```

I do not recall ever seeing the dialog to exit setup...

**Comment by Vogel612** [comment]
Thanks for reporting this. Just for my understanding: You did not see the insteller do anything? Or is this the same as https://github.com/rubberduck-vba/Rubberduck/issues/3921?

**Comment by yosso** [comment]
There does not appear to be an entry in the start menu for "Rubber Duck".  Additionally, Rubber Duck does not show up in the uninstall programs.

**Comment by bclothier** [comment]
@yosso is it consistently doing that? I would expect you to see a dialog, especially that the log says that user chose `Yes`, after a bit of pause.

**Comment by yosso** [comment]
@bclothier Well, I was going to say yes, but this time the log file was not created?

**Comment by bclothier** [comment]
@yosso  pardon me if that's a dumb question but could it be that it's not showing because you are searching for `Rubber Duck`, not `Rubberduck`? 

If you were browsing and did not find the `Rubberduck` at all, then can you confirm if the directory for Rubberduck exists in fact?

**Comment by retailcoder** [comment]
The local admin account allowed you to correctly uninstall the previous version, so there's that :+1: 

I have local admin privs on my work laptop too, and can confirm that installing isn't as straightforward as it should be: you'll have to run the "repair registration" step from the start menu after setup completes.

That said, per the logs it appears installation prompted to exit, and proceeded to cancel - which is consistent with Rubberduck not ending up installed. If that dialog never actually showed up, it looks like we have a rather weird bug in there that needs further investigation.

Since the old admin-required install is gone and the installer won't allow per-user installation given local admin privs (I have the same problem.. although all-users install *did* work here...), I'd suggest removing local admin privs (you'll need to log out and then log back in once IT completes this) - then you should be able to install for your user only, and registration should "just work".

Sorry for the inconveniences... installers are *hard*!

**Comment by bclothier** [comment]
I just missed one thing in your OP - you said `Right-click and choose Run as admin`. But that's not required! Try and running it normally without admin and see if it works.

**Comment by yosso** [comment]
I have been installing via the right-click and the login with my local admin account.

I did just do another uninstall (local admin) and then another install as a non-admin for user account only.  This time there is a Rubber Duck entry in the start menu.  

I did select the update registry script, it opened a cmd window and closed.

Ran Excel and received an error dialog about Rubberduck's startup sequence throwing an unexpected exception.  Please check the logs for more issues....

Didn't see any log files in the local\Rubberduck folder?

The temp folder does have a log file for the install, and it seems to show that everything has installed correctly.

Might do a reboot and try again on my break. 

@retailcoder - So very true (regarding installers) :-)

**Comment by retailcoder** [comment]
IIRC logging is still opt-in, so it would have to be configured/enabled first... which is annoying if RD won't even start. There should be an XML configuration file under `%program%\Rubberduck` (or whatever the install folder was), that defines all the default configuration settings (file would be `Rubberduck.Core.dll.config`); there's a node named `GeneralSettings`:

```xml
          <GeneralSettings xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
            <Language Code="en-US" />
            <CanShowSplash>true</CanShowSplash>
            <CanCheckVersion>true</CanCheckVersion>
            <CompileBeforeParse>true</CompileBeforeParse>
            <IsSmartIndenterPrompted>false</IsSmartIndenterPrompted>
            <IsAutoSaveEnabled>false</IsAutoSaveEnabled>
            <AutoSavePeriod>10</AutoSavePeriod>
            <MinimumLogLevel>6</MinimumLogLevel>
            <EnableExperimentalFeatures />
          </GeneralSettings>
```

Change the `MinimumLogLevel` to `0`, save the file, and try to reload RD - if it still crashes, there should now be a log file under `%appdata%\Rubberduck\Logs` with something to chew on.

**Comment by bclothier** [comment]
FYI the script wasn't necessary if you ran as per-user install. It's mainly there for per-machine install where each user must manually register for their profile though it can be also used to repair the registration.

Also, just as a FYI - if you ran the installer non-admin, you have the option of installing per-machine still, and it would have ran the installer (elevated) separately and ALSO registered automatically. 

I believe we have a open issue #3861 about that kind of issue which need to be resolved. Meanwhile, you could try and create the config manually:

In `%appdata%\Rubberduck\` folder, create a file `rubberduck.config` and save the following text:

```
<?xml version="1.0" encoding="utf-8"?>
<Configuration>
  <GeneralSettings>
    <Language Code="en-US" />
    <CanShowSplash>true</CanShowSplash>
    <CanCheckVersion>true</CanCheckVersion>
    <CompileBeforeParse>true</CompileBeforeParse>
    <IsSmartIndenterPrompted>false</IsSmartIndenterPrompted>
    <IsAutoSaveEnabled>false</IsAutoSaveEnabled>
    <AutoSavePeriod>10</AutoSavePeriod>
    <MinimumLogLevel>0</MinimumLogLevel>
    <EnableExperimentalFeatures />
  </GeneralSettings>
</Configuration>
```

That will enable trace level logging and try again.

**Comment by retailcoder** [comment]
@bclothier right. Creating a configuration file manually should work too. My approach would have been to just change the default log setting to make it opt-out :smile:

**Comment by retailcoder** [comment]
note to self: making logging opt-out could be as simple as changing the ordering of log levels enum members so that `NoLogging` has the much more intuitive value `0`.

**Comment by yosso** [comment]
Enabled logging (via editing the Rubberduck.Core.dll.config), after doing an install.  

Still not seeing any log files in C:\ProgramData\Rubberduck.

**Comment by bclothier** [comment]
It doesn't go there. It'll be in `%APPDATA%\Rubberduck\Logs`

**Comment by Vogel612** [comment]
the logfiles should be under `%APPDATA%/Rubberduck`

**Comment by yosso** [comment]
Sorry about that. :-(.

#Log Dump

2018-04-11 10:38:35.5505;FATAL-;Rubberduck._Extension;Startup sequence threw an unexpected exception.;System.ArgumentException: An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at Rubberduck.VBEditor.Events.VBEEvents.RegisterComponents(IVBProject project) in C:\projects\rubberduck\Rubberduck.VBEEditor\Events\VBEEvents.cs:line 81
   at Rubberduck.VBEditor.Events.VBEEvents..ctor(IVBE vbe) in C:\projects\rubberduck\Rubberduck.VBEEditor\Events\VBEEvents.cs:line 63
   at Rubberduck.VBEditor.Events.VBEEvents.Initialize(IVBE vbe) in C:\projects\rubberduck\Rubberduck.VBEEditor\Events\VBEEvents.cs:line 22
   at Rubberduck.Root.RubberduckIoCInstaller.RegisterConstantVbeAndAddIn(IWindsorContainer container) in C:\projects\rubberduck\Rubberduck.Main\Root\RubberduckIoCInstaller.cs:line 814
   at Rubberduck.Root.RubberduckIoCInstaller.Install(IWindsorContainer container, IConfigurationStore store) in C:\projects\rubberduck\Rubberduck.Main\Root\RubberduckIoCInstaller.cs:line 86
   at Castle.Windsor.WindsorContainer.Install(IWindsorInstaller[] installers, DefaultComponentInstaller scope)
   at Castle.Windsor.WindsorContainer.Install(IWindsorInstaller[] installers)
   at Rubberduck._Extension.Startup() in C:\projects\rubberduck\Rubberduck.Main\Extension.cs:line 221
2018-04-11 10:38:35.5845;FATAL-;Rubberduck._Extension;System.Exception: Rubberduck's startup sequence threw an unexpected exception. Please check the Rubberduck logs for more information and report an issue if necessary
   at Rubberduck._Extension.Startup() in C:\projects\rubberduck\Rubberduck.Main\Extension.cs:line 234
   at Rubberduck._Extension.InitializeAddIn() in C:\projects\rubberduck\Rubberduck.Main\Extension.cs:line 189;System.Exception: Rubberduck's startup sequence threw an unexpected exception. Please check the Rubberduck logs for more information and report an issue if necessary
   at Rubberduck._Extension.Startup() in C:\projects\rubberduck\Rubberduck.Main\Extension.cs:line 234
   at Rubberduck._Extension.InitializeAddIn() in C:\projects\rubberduck\Rubberduck.Main\Extension.cs:line 189
2018-04-11 10:38:41.0186;INFO-;Rubberduck._Extension;Rubberduck is shutting down.;
2018-04-11 10:38:41.0186;INFO-;Rubberduck._Extension;No exceptions were thrown.;

**Comment by bclothier** [comment]
Hmm. That looks funny to me.

Do me a favor, open this in another Office application. Maybe Word? Does it work?

What about a different Excel spreadsheet?

What about Excel with all its add-ins disabled?

**Comment by yosso** [comment]
Word did not throw an exception.

Turned off all of my Excel add-ins.

No exceptions! Yay!

*Turned on my ADSK Vault COM Addin
**No exception.

*Turned on my MySQL COM Addin
**No exception.

*Turned on one of my personal VBA Addins.
**No exception

*Turned on VBA Code Documenter 5.0 (Rob Bovey)
**No Exception

*Turned on a commercial Addin (XLC Addin for Excel)
*Turned on AMA Addin (another personal addin)
**No Exception

*Turned on VBA Code Cleaner 5.0
**No Exception

*Turned on Add-in Express Toys VB.net & PDF Xchange
**No Exception

There were a couple of other addins that I had placed at one time or another when attempting to create an Excel addin with Visual Studio.  I didn't re-activate those addins yet.

**Comment by bclothier** [comment]
From the log, I see there is an error with setting up the event listener for one of `VBProject`s -- the log says that it's being added with same identifier twice, which *shouldn't* be possible but obviously that assumption got violated somehow.

Do you happen to know if any of your add-ins has a help file? If so, do more than one project point to the same help file?

**Comment by retailcoder** [comment]
@bclothier depending on how that's wired-up, it's actually rather easy to blow that assumption: are you using the project name or the project ID? Every VBA project has the same default name, that shouldn't be your key here...

**Comment by retailcoder** [comment]
Nevermind, ..I should have just gone and checked :+1: 

    _components.Add(project.ProjectId, components);

A key collision here is indeed *theoretically* impossible.

**Comment by yosso** [comment]
I always _try_ to rename my projects....but...you never know.

I've never implemented an official help file, that I can recall...

**Comment by bclothier** [comment]
If you can reproduce the exception by enabling the remaining add-ins, then you would need to check its `HelpFile` property and see whether it contains a value that's already used by other VBA project/add-in. (Note that RD will put in a numeric value so as long they are distinct, no problem).

**Comment by yosso** [comment]
Will do.  All of the critical ADDIN's have be added back into place and no exceptions have been thrown.

There was way too much detritus  being loaded from all of my "experimentation". 

One other data point...my previous install of RD wasn't working in my AutoDesk Inventor VBA IDE, but this time around it appears to be loading into Inventor VBA IDE.

**Comment by retailcoder** [comment]
@yosso  thanks for the update. FWIW I have Rubberduck 2.2 release loading fine in a Sage 300 ERP local install (local admin), and refusing to load in Sage 300 ERP with a per-user install through remote desktop - and that install works fine in Excel... that's what prompted the "may not load in all hosts" comment in the release notes.

**Comment by bclothier** [comment]
Since this issue has 2 separate problems, one which is going to be addressed by upcoming PR #3948 and another not related to installer, I'm opening a separate issue and closing this issue.
